core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,116 +288,3206 @@ def validate(cls, item):
+        """
+        if len(item.id) != 12:
+            cls.errors['model id length check'] = (
+                'Model id %s: Model id should be of length 12 but instead has '
+                'length %s' % (item.id, len(item.id)))
+
+    @classmethod
+    def _validate_sent_datetime(cls, item):
+        """Validate that sent_datetime of model is less than current time.
+
+        Args:
+            item: BulkEmailModel to validate.
+        """
+        current_datetime = datetime.datetime.utcnow()
+        if item.sent_datetime > current_datetime:
+            cls.errors['sent datetime check'] = (
+                'Model id %s: The sent_datetime field has a value %s which is '
+                'greater than the time when the job was run'
+                ) % (item.id, item.sent_datetime)
+
+    @classmethod
+    def _validate_sender_email(cls, item):
+        """Validate that sender email corresponds to email of user obtained
+        by using the sender_id.
+
+        Args:
+            item: BulkEmailModel to validate.
+        """
+        _, sender_models = (cls.external_models['sender_id'])
+        sender_model = sender_models[0][1]
+        if sender_model and not sender_model.deleted:
+            if sender_model.email != item.sender_email:
+                cls.errors['sender email check'] = (
+                    'Model id %s: Sender email %s in model does not match '
+                    'with email %s of user obtained through sender id') % (
+                        item.id, item.sender_email, sender_model.email)
+
+    @classmethod
+    def _get_validation_functions(cls):
+        return [
+            cls._validate_id_length, cls._validate_sent_datetime,
+            cls._validate_sender_email]
+
+
+class GeneralFeedbackEmailReplyToIdModelValidator(BaseModelValidator):
+    """Class for validating GeneralFeedbackEmailReplyToIdModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'item.id.user_id': (
+                user_models.UserSettingsModel, [
+                    item.id[:item.id.find('.')]]),
+            'item.id.thread_id': (
+                feedback_models.GeneralFeedbackThreadModel, [
+                    item.id[item.id.find('.') + 1:]]),
+        }
+
+    @classmethod
+    def _validate_reply_to_id_length(cls, item):
+        """Validate that reply_to_id length is less than or equal to
+        REPLY_TO_ID_LENGTH.
+
+        Args:
+            item: GeneralFeedbackEmailReplyToIdModel to validate.
+        """
+        if len(item.reply_to_id) > email_models.REPLY_TO_ID_LENGTH:
```

Comment:
Add a comment explaining the check here i.e. why should this condition hold?

Reply:
Done.